### PR TITLE
Add types for appium-adb

### PR DESCRIPTION
**Pull request by NotWoods**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/appium-adb/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Type definitions for appium-adb 8.16
+// Project: https://github.com/appium/appium-adb
+// Definitions by: Tiger Oakes <https://github.com/NotWoods>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export * from './lib/adb';
+export { default } from './lib/adb';
```
Comment (peterblazejewicz) [inline types/appium-adb/index.d.ts:7]:
```suggestion
import {
  ADB,
  DEFAULT_ADB_PORT,
  getAndroidBinaryPath,
  getSdkRootFromEnv
} from './lib/adb';

export default ADB;
export { ADB, DEFAULT_ADB_PORT, getAndroidBinaryPath, getSdkRootFromEnv };
```
notice, this is just same as in source (ok, nearly)

File: types/appium-adb/lib/logcat.d.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    /**
+     * Series of <tag>[:priority]
+     * where <tag> is a log component tag (or * for all) and priority is:
+     *  V    Verbose
+     *  D    Debug
+     *  I    Info
+     *  W    Warn
+     *  E    Error
+     *  F    Fatal
+     *  S    Silent (supress all output)
```
Comment (peterblazejewicz) [inline types/appium-adb/lib/logcat.d.ts:19]:
```suggestion
     *  S    Silent (suppress all output)
```

File: types/appium-adb/lib/tools/android-manifest.d.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+     * particular package.
+     *
+     * @param manifest - Full path to the precompiled manifest
+     *                            created by `compileManifest` method call
+     *                            without .apk extension
+     * @param srcApk - Full path to the existing valid application package, where
+     *                          this manifest has to be insetred to. This package
+     *                          will NOT be modified.
+     * @param dstApk - Full path to the resulting package.
+     *                          The file will be overriden if it already exists.
```
Comment (peterblazejewicz) [inline types/appium-adb/lib/tools/android-manifest.d.ts:64]:
```suggestion
     *                          The file will be overridden if it already exists.
```

File: types/appium-adb/lib/tools/android-manifest.d.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    /**
+     * Replace/insert the specially precompiled manifest file into the
+     * particular package.
+     *
+     * @param manifest - Full path to the precompiled manifest
+     *                            created by `compileManifest` method call
+     *                            without .apk extension
+     * @param srcApk - Full path to the existing valid application package, where
+     *                          this manifest has to be insetred to. This package
```
Comment (peterblazejewicz) [inline types/appium-adb/lib/tools/android-manifest.d.ts:61]:
```suggestion
     *                          this manifest has to be inserted to. This package
```

File: types/appium-adb/lib/tools/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    AvdLaunchOptions,
+    BinaryVersion,
+    BridgeVersion,
+    Version,
+    RootResult,
+    BinaryName,
+} from './system-calls';
+import apkSigningMethods, { CertCheckOptions, KeystoreHash } from './apk-signing';
+import apkUtilsMethods, {
+    APP_INSTALL_STATE,
```
Comment (peterblazejewicz) [inline types/appium-adb/lib/tools/index.d.ts:17]:
```ts
APP_INSTALL_STATE
```
those one and other in that file are imported but never exported

File: types/appium-adb/index.d.ts (hunk 1 above)
Comment (NotWoods) [reply types/appium-adb/index.d.ts:7]: Is it preferred to not export the interfaces?

Comment (peterblazejewicz) [reply types/appium-adb/index.d.ts]: oh, no, the only concern here was to not export everything, say types that are not needed to use package itself and are added by TS migration. So it's perfectly ok to export types e.g. that way:
```ts
// Type definitions for appium-adb 8.16
// Project: https://github.com/appium/appium-adb
// Definitions by: Tiger Oakes <https://github.com/NotWoods>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped

import {
    ADB,
    DEFAULT_ADB_PORT,
    getAndroidBinaryPath,
    getSdkRootFromEnv,
    AdbExecutable,
    CreateAdbOptions,
} from './lib/adb';

export default ADB;
export {
    ADB, 
    DEFAULT_ADB_PORT,
    getAndroidBinaryPath,
    getSdkRootFromEnv,
    AdbExecutable,
    CreateAdbOptions
};
```
(list every interface once can find useful or even required when using those types).
Sorry for the hassle. 
I"ll accept, you're free to amend, thx!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@NotWoods thx!

File: types/appium-adb/index.d.ts (hunk 1 above)
Comment (NotWoods) [reply types/appium-adb/index.d.ts]: Updated so the index file imports from tools directly like this!

**Comment by NotWoods** [comment]
Ready to merge

**Review by peterblazejewicz — APPROVED** [review]

**Comment by NotWoods** [comment]
Ready to merge

**Pull request merged**